### PR TITLE
Allow build with Qt6

### PR DESCRIPTION
**Pull request by jmlich**
todo:
- [x] mlite related commit d8297563c7c2ba0b161418bbba7b9db27f245650 should be probably removed or is it acceptable for upstream?
- [x] fromTime_t(), toTime_t(), setTime_t() should probably contain #ifdef
- [x] need to check if from/to/setTime_t date type rages are okay. Eventually change to qint64
- [x] need to check when enableTestMode -> setTestModeEnabled was done and add #ifdef eventually

### CODE REVIEW

**Comment by jmlich** [comment]
I believe the overflow when int is used instead of qint64 for is not necessary for now.

File: src/lib/vkpostsdatabase.cpp [hunk 1]
```diff
@@ -297,43 +297,43 @@ VKPostsDatabase::Post::Ptr VKPostsDatabase::Post::create(const SocialPost::Const
 
     const QVariantMap &extra = socialPost->extra();
 
-    vkPost->fromId = extra.value(POST_FROM_ID_KEY).toInt();
-    vkPost->toId = extra.value(POST_TO_ID_KEY).toInt();
-    vkPost->replyOwnerId = extra.value(POST_REPLY_OWNER_ID_KEY).toInt();
-    vkPost->replyPostId = extra.value(POST_REPLY_POST_ID_KEY).toInt();
-    vkPost->friendsOnly = extra.value(POST_FRIENDS_ONLY_KEY).toBool();
-    vkPost->signerId = extra.value(POST_SIGNER_ID_KEY).toInt();
+    vkPost->fromId = QVariant(extra.value(POST_FROM_ID_KEY)).toInt();
```
Comment (pvuorela) [inline src/lib/vkpostsdatabase.cpp:300]: Hm, not following. value() on QVariantMap should already return a QVariant?

File: src/qml/qml.pro [hunk 2]
```diff
@@ -84,7 +84,9 @@ QMAKE_EXTRA_TARGETS += qmltypes
 TS_FILE = $$OUT_PWD/socialcache.ts
 EE_QM = $$OUT_PWD/socialcache_eng_en.qm
 
-ts.commands += lupdate $$PWD/.. -ts $$TS_FILE
+qtPrepareTool(LUPDATE, lupdate)
```
Comment (pvuorela) [inline src/qml/qml.pro:87]: Hm, why is this needed? Don't recall seeing too much usage of qtPrepareTool, not finding documentation and so far it's been working ok just telling qmake what commands to execute.

**Review by pvuorela — COMMENTED** [review]
Mostly seems ok, but couple small comments.

File: src/qml/qml.pro (hunk 2 above)
Comment (jmlich) [reply src/qml/qml.pro:87]: the issue is that some OS provides lupdate-qt5, some lupdate-qt6, and others just lupdate. Also, the lupdate can be in different path e.g. /opt/qt5

Comment (pvuorela) [reply src/qml/qml.pro:87]: Ah right. Fair enough then, if the call handles varying binary names and locations.

File: src/lib/vkpostsdatabase.cpp (hunk 1 above)
Comment (jmlich) [reply src/lib/vkpostsdatabase.cpp:300]: I guess, I wasn't able to compile it without that. I need to check it again

**Comment by pvuorela** [comment]
Not to get stuck on the QVariantMap, I pushed all the other changes now. Closing this now. If any further need, can open new PRs.
